fails to build as a subproject (ld: cannot find -ljwt)

I am having a very simple setup:

```
libjwt/         <-- fresh unmodified git clone (master)
CMakeLists.txt  <-- my cmake file
jwttest.c       <-- a very simple test program
```
Contents of CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.5.1)

project(myproject C)

SET(BUILD_TESTS                  FALSE CACHE BOOL "")
SET(BUILD_SHARED_LIBS            TRUE  CACHE BOOL "")
SET(WITHOUT_OPENSSL              FALSE CACHE BOOL "")
SET(ENABLE_DEBUG_INFO_IN_RELEASE FALSE CACHE BOOL "")

add_subdirectory(libjwt)

include_directories(libjwt/include)
add_executable(jwttest jwttest.c)
target_link_libraries(jwttest jwt)
```

Contents of jwttest.c:

```
#include "jwt.h"

int main() {
  jwt_t *jwt = NULL;
  jwt_new(&jwt);
  return 0;
}
```

Now when I am running cmake on this directory, and later make, it fails:

```
$ make
Scanning dependencies of target jwttest
[ 16%] Building C object CMakeFiles/jwttest.dir/jwttest.c.o
[ 33%] Linking C executable jwttest
/usr/bin/ld: cannot find -ljwt
collect2: error: ld returned 1 exit status
CMakeFiles/jwttest.dir/build.make:83: recipe for target 'jwttest' failed
make[2]: *** [jwttest] Error 1
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/jwttest.dir/all' failed
make[1]: *** [CMakeFiles/jwttest.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

```

Is this intended?

--- a/libjwt/CMakeLists.txt
+++ b/libjwt/CMakeLists.txt
@@ -1,4 +1,4 @@
-set (TARGET_NAME ${CMAKE_PROJECT_NAME})
+set (TARGET_NAME ${PROJECT_NAME})
 
 option (BUILD_SHARED_LIBS "Build libjwt as shared library instead as static one." OFF)
 option (WITHOUT_OPENSSL "Use GnuTLS for encryption instead of OpenSSL" OFF)
@@ -87,6 +87,6 @@ install (TARGETS ${TARGET_NAME}
 	)
 
 install (FILES
-	${CMAKE_SOURCE_DIR}/include/jwt.h
+	${PROJECT_SOURCE_DIR}/include/jwt.h
 	DESTINATION include/
 	)

